src/Microsoft.Graph.Core/Extensions/TokenCredentialExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+
+namespace Microsoft.Graph
+{
+    using Microsoft.Identity.Client;
+    using System;
+    using System.Net.Http.Headers;
+    using Azure.Core;
+
+    internal static class TokenCredentialExtensions
+    {
+        /// <summary>
+        /// Gets retry after timespan from <see cref="RetryConditionHeaderValue"/>.
+        /// </summary>
+        /// <param name="tokenCredential">n <see cref="TokenCredential"/> object.</param>
+        /// <param name="serviceException">A <see cref="MsalServiceException"/> with RetryAfter header</param>
+        internal static TimeSpan GetRetryAfter(this TokenCredential tokenCredential, MsalServiceException serviceException)
+        {
+            RetryConditionHeaderValue retryAfter = serviceException.Headers?.RetryAfter;
+            TimeSpan? delay = null;
```

Comment:
```suggestion
            TimeSpan? delay = retryAfter?.Delta ?? retryAfter?.Date?.Value?.Offset;
```
and remove the if/else if blocks below

Reply:
Resolved via #274